src/client/components/Room/Room.jsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useRef, useState } from 'react';
+import styles from './room.module.css';
+
+export const Room = ({ room, socket }) => {
+  const [editing, setEditing] = useState(false);
+  const [newName, setNewName] = useState(room);
+
+  const controller = useRef(null);
+
+  useEffect(() => {
+    const handleClickOutside = (evt) => {
+      if (controller.current && !controller.current.contains(evt.target)) {
+        setEditing(false);
+
+        const newRoomName = newName.trim();
+
+        if (newRoomName !== room) {
+          const action = { type: 'rename', room, newRoomName };
+
+          socket.send(JSON.stringify(action));
+        }
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [room, socket, newName]);
+
+  const onClickHandlerDelete = () => {
+    const action = { type: 'remove', room };
+
+    socket.send(JSON.stringify(action));
```

Comment:
Similarly, before sending the remove action, ensure the socket is open by checking `socket.readyState === WebSocket.OPEN`.